crates/fuel-streams-publisher/src/publisher.rs

```diff
@@ -227,7 +227,7 @@ impl Publisher {
         Ok(())
     }
 
-    pub async fn run(mut self) -> anyhow::Result<Self> {
+    pub async fn run(&mut self) -> anyhow::Result<()> {
```

Comment:
@lostman I think the idea here is (and this is not my code btw.) that once you create an instance of the Publisher, you can call .run() on it and it would consume self meaning you would not be allowed to run any other methods once you call run on it. I personally prefer this design. Of course returning self is not great here, so you changes to () are fine. Nevertheless ,I feel the signatrure argument should be as before pub async fn run(mut self)

Reply:
@Jurshsmith @pedronauck  ?

Reply:
Exactly 👍🏾 @0xterminator. The old signature explicitly consumes(or moves) the publisher to remove the ambiguity of reactive mutations after running.